plugins/search-backend-module-pg/src/PgSearchEngine/PgSearchEngine.ts

```diff
@@ -26,28 +26,112 @@ import {
   DatabaseStore,
   PgSearchQuery,
 } from '../database';
+import { v4 as uuid } from 'uuid';
+import { Config } from '@backstage/config';
 
+/**
+ * Search query that the Postgres search engine understands.
+ * @public
+ */
 export type ConcretePgSearchQuery = {
   pgQuery: PgSearchQuery;
   pageSize: number;
 };
 
+/**
+ * Options available for the Postgres specific query translator.
+ * @public
+ */
+export type PgSearchQueryTranslatorOptions = {
+  highlightOptions: PgSearchHighlightOptions;
+};
+
+/**
+ * Postgres specific query translator.
+ * @public
+ */
+export type PgSearchQueryTranslator = (
+  query: SearchQuery,
+  options: PgSearchQueryTranslatorOptions,
+) => ConcretePgSearchQuery;
+
+/**
+ * Options to instantiate PgSearchEngine
+ * @public
+ */
+export type PgSearchOptions = {
+  config: Config;
+  database: PluginDatabaseManager;
+};
+
+/**
+ * Options for highlighting search terms
+ * @public
+ */
+export type PgSearchHighlightOptions = {
+  useHighlight?: boolean;
+  maxWords?: number;
+  minWords?: number;
+  shortWord?: number;
+  highlightAll?: boolean;
+  maxFragments?: number;
+  fragmentDelimiter?: string;
+  preTag: string;
+  postTag: string;
+};
+
 export class PgSearchEngine implements SearchEngine {
-  constructor(private readonly databaseStore: DatabaseStore) {}
+  private readonly highlightOptions: PgSearchHighlightOptions;
+  constructor(private readonly databaseStore: DatabaseStore, config: Config) {
```

Comment:
While you're touching this, it might be good to perhaps mark this constructor `@deprecated` too, so that we can mark it private in a future release.

Reply:
This has been added too 👍